src/services/expenses.service.js

```diff
@@ -0,0 +1,67 @@
+const { getNumberId } = require('./../utils');
+
+class Expenses {
+  constructor(initialExpenses) {
+    this._expenses = initialExpenses || [];
+  }
+
+  getAll({ userId, categories, from, to }) {
+    let res = this._expenses;
+
+    if (userId) {
+      res = res.filter((expense) => expense.userId === userId);
+    }
+
+    if (categories) {
+      res = res.filter((expense) => {
+        return categories.includes(expense.category);
+      });
+    }
+
+    if (from) {
+      const fromDate = new Date(from);
+
+      res = res.filter((expense) => {
+        return new Date(expense.spentAt) > fromDate;
+      });
+    }
+
+    if (to) {
+      const toDate = new Date(to);
+
+      res = res.filter((expense) => new Date(expense.spentAt) < toDate);
```

Comment:
Consider using `<=` for the `to` date comparison if you want to include expenses on the `to` date.